### DISCUSSION TITLE
Is it possible to output the original, untransformed or raw content when generating files?

### DISCUSSION THREAD

**Discussion by mrmartineau** [11ty/eleventy#2769] [Q&A] [answered]
I am trying to generate a JSON file with my content, something like this: 

```njk
---
permalink: /search.json
---

[
	{% for note in collections.notes %}
		{
			"title": "{{ note.data.title }}",
			"date": "{{ note.date }}",
			"url": "{{ note.url }}",
			"content": "{{ note.templateContent }}"
		}{% if not loop.last %},{% endif %}
	{% endfor %}
]
```

But I don't want the `templateContent ` property to be transformed? I'd like to return the original markdown. Is that possible?

FYI I am using the new 11ty v2 beta

**Comment by pdehaan** [comment]
Possibly a custom filter:

```js
eleventyConfig.addFilter("dumpColl", function (coll = [], indent = 2) {
  const res = coll.map(item => {
    return {
      title: item.data.title,
      date: item.date,
      url: item.url,
      content: item.template._frontMatter.content.trim(),
    };
  });
  return JSON.stringify(res, null, indent);
});
```

### src/notes/one.md
```md
---
title: Page One
---

# {{ title }}
[Two](/two/)
```

### src/search.njk
```njk
---
permalink: /search.json
---

{{- collections.notes | dumpColl | safe -}}
```

### OUTPUT
```json
[
  {
    "title": "Page One",
    "date": "2023-01-25T01:37:57.578Z",
    "url": "/notes/one/",
    "content": "# {{ title }}\n[Two](/two/)"
  },
  {
    "title": "Page Two",
    "date": "2023-01-25T01:38:55.685Z",
    "url": "/notes/two/",
    "content": "# {{ title }}\n[Three](/three/)"
  },
  {
    "title": "Page Three",
    "date": "2023-01-25T01:39:33.308Z",
    "url": "/notes/three/",
    "content": "# {{ title }}\n[One](/one/)"
  }
]
```

**Reply by mrmartineau** [reply]
@pdehaan thanks for the suggestion. I hadn't thought of making a custom filter like that. It also gives me the idea that I could use a custom shortcode as well. Lots of food for thought. Cheers

**Reply by mrmartineau** [reply]
I created a shortcode that to build what I need:

```js
eleventyConfig.addShortcode('algoliaIndex', function (notes) {
    const algoliaIndex = notes.map((note) => {
      return {
        title: note.data.title,
        url: note.url,
        date: note.date,
        content: note.template.frontMatter.content,
        tags: note.data?.tags?.length ? note.data.tags : [],
      }
    })
    return JSON.stringify(algoliaIndex)
  })
```

and how its used in the template:

```njk
---
permalink: /algolia.json
---

{% algoliaIndex collections.notes %}
```

**Comment by nhoizey** [comment]
@pdehaan is `item.template._frontMatter.content` safe to use? Won't it disapear without notice in the future, as it's an internal data?

I think there was an issue asking for a better exposition of source Markdown, but I can't find it.

**Reply by aloxe** [reply]
@nhoizey I've been using `item.templateContent` I don't know which one is the safest.

**Reply by pdehaan** [reply]
I don't imagine anything prefixed w/ an [undocumented] underscore is guaranteed. It's basically just hacking into the [gray-matter](https://www.npmjs.com/package/gray-matter) properties.

https://github.com/11ty/eleventy/issues/1206#issuecomment-633559519 implies something like `{{ post.template.frontMatter.content }}` might work too.

**Reply by mrmartineau** [reply]
@pdehaan FYI `post.template.frontMatter.content` seems to return the same as `item.template._frontMatter.content` and both are what I need to build my Algolia search index.

**Comment by zachleat** [comment]
Upvotes welcome/subscribes recommended for https://github.com/11ty/eleventy/issues/1206

**Reply by nhoizey** [reply]
That's the issue I remembered but couldn't find again! 👍

**Comment by mrmartineau** [comment, marked as answer]
I created a shortcode that to build what I need:

```js
eleventyConfig.addShortcode('algoliaIndex', function (notes) {
    const algoliaIndex = notes.map((note) => {
      return {
        title: note.data.title,
        url: note.url,
        date: note.date,
        content: note.template.frontMatter.content,
        tags: note.data?.tags?.length ? note.data.tags : [],
      }
    })
    return JSON.stringify(algoliaIndex)
  })
```

and how its used in the template:

```njk
---
permalink: /algolia.json
---

{% algoliaIndex collections.notes %}
```

Thanks to @pdehaan for [leading me](https://github.com/11ty/eleventy/discussions/2769#discussioncomment-4782403) to the solution I wanted
